File: 243__Shortest_Word_Distance.py
class Solution(object):
    def shortestDistance(self, words, word1, word2):
        """
        :type words: List[str]
        :type word1: str
        :type word2: str
        :rtype: int
        """
        result = len(words) + 1
        index = None
        found = None
        for i, word in enumerate(words):
            if word == word1 or word == word2:
                if found is not None:
                    if word != found:
                        result = min(result, i - index)
                found = word
                index = i
        return result

File: Java/243__Shortest_Word_Distance.java
class Solution {
    public int shortestDistance(String[] words, String word1, String word2) {
        int pos1 = -1, pos2 = -1;
        int result = Integer.MAX_VALUE;
        for (int i = 0; i < words.length; i++) {
            if (words[i].equals(word1)) pos1 = i;
            if (words[i].equals(word2)) pos2 = i;
            if (pos1 != -1 && pos2 != -1)
                result = Math.min(result, Math.abs(pos1 - pos2));
        }
        return result;
    }
}
